src/app/pages/my-openstax/main-card/navigator/navigator.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import NavigationContext from './navigation-context';
+import {WalkthroughCookieContextProvider} from './walkthrough/walkthrough-cookie-context';
+import cn from 'classnames';
+import Walkthrough from './walkthrough/walkthrough';
+import './navigator.scss';
+
+function useElById(id) {
+    const [el, setEl] = React.useState(document.getElementById(id));
+
+    const elPoll = window.setInterval(() => {
+        if (el) {
+            window.clearInterval(elPoll);
+        } else {
+            setEl(document.getElementById(id));
+        }
+    }, 40);
+
+    return el;
+}
```

Comment:
I think should this might need:
  * To be enclosed in a useEffect that returns a fn to clear the interval on unmount
  * have a timeout setting to set an upper bound on how long it'll keep retrying.  That way if the element is never found it'll stop at some point